### DISCUSSION TITLE
Question about speed on animevideov3

### DISCUSSION THREAD

**Discussion by Aleonz00** [xinntao/Real-ESRGAN#372] [Q&A]
NCNN usually upscales 480p images at a speed of 970 images per minute (16fps?) with "realesr-animevideov3-x2" on my Titan Xp. I can upscale hundreds of thousands of images in what I thought was a very short amount of time.

Then I found your comparison page which says your Nvidia V100 GPU can upscale 480p at 65.9 frames per SECOND on PyTorch. Am I reading this wrong? I get that the V100 is better than the Titan XP, but when I tried upscaling the same images using the Python script, it was ONE image per FEW SECONDS (Testing Image 1, Testing Image 2, etc). Extremely slow using the same model and same scaling in comparison to NCNN which seems to be considered by the community as much slower than Python. But that's not what I've ever experienced. Even with v2, Python was just as slow (1 image per 3 or so seconds), while v2 NCNN was upscaling at 450 images per minute. 

I'm using the latest RealESRGAN master for Python animevideov3.

"python inference_realesrgan.py -n realesr-animevideov3 -s 2 -i input -o output" is the CMD paste that I'm using for your reference.

I was completely satisfied with NCNN's speed, but if I can get Python to be faster than NCNN, I'd like to know what's wrong.

**Comment by ToTheBeginning** [comment]
To focus more on the model itself, as mentioned in [this line](https://github.com/xinntao/Real-ESRGAN/blob/e2db57602068a839b0f53df9efca7e57fa3c8a99/docs/anime_comparisons.md?plain=1#L56), the [table](https://github.com/xinntao/Real-ESRGAN/blob/e2db57602068a839b0f53df9efca7e57fa3c8a99/docs/anime_comparisons.md?plain=1#L58) only  reports the model `forward` time and ignores the IO time. Note that IO time can take up a significant portion of the total inference time.

For pytorch/python users, especially those with multiple GPUs, we recommend using `inference_realesrgan_video ` scipt for inference video or frames. You can find some example usages [here](https://github.com/xinntao/Real-ESRGAN/blob/e2db57602068a839b0f53df9efca7e57fa3c8a99/docs/anime_video_model.md?plain=1#L40).

**Comment by FlM122** [comment]
which model is the best for anime videos??? to ensure that the quality is as good as possible and that the speed is high . I found AnimeJaNai, but it makes the contour lines too sharp.
